example/js/js_image_layer.bzl

```diff
@@ -0,0 +1,260 @@
+with open(manifest_path, "r") as manifest_fp:
+    manifest = json.load(manifest_fp)
+def strip_execroot(p):
+    parts = p.split(os.sep)
+    i = parts.index("execroot") 
+    return os.sep.join(parts[i+2:]) 
+def get_runfiles_path(p):
+    for entry in manifest:
+        if entry[2] == p:
+            return entry[1]
+    raise Exception("could not find a corresponding file for %s within manifest.")
+for entry in manifest:
+    p = entry[2]
+    if "node_modules" in p and os.path.islink(p):
+        link_to = os.path.realpath(p)
+        link_to_execroot_stripped = strip_execroot(link_to)
+        overwritten_to = get_runfiles_path(link_to_execroot_stripped)
+        # fix it!
+        entry[0] = 1 # make it a symlink
+        entry[2] = "/%s" % overwritten_to
+        entry[3] = "0777"
+os.remove(manifest_path)
+with open(manifest_path, "w") as manifest_w:
+    manifest_w.write(json.dumps(manifest))
+r = subprocess.run([build_tar_path] + sys.argv[1:])
+sys.exit(r.returncode)
+"""
+
+def js_image_layer(name, binary, root = None, **kwargs):
+    """Creates two tar files `:<name>/app.tar` and `:<name>/node_modules.tar`
+
+    Final directory tree will look like below
+    /{root of js_image_layer}/{package_name() if any}/{name of js_binary}.sh -> entrypoint
+    /{root of js_image_layer}/{package_name() if any}/{name of js_binary}.sh.runfiles -> runfiles directory (almost identical to one bazel lays out)
+    Args:
+        name: name for this target. Not reflected anywhere in the final tar.
+        binary: label to js_image target
+        root: Path where the js_binary will reside inside the final container image.
+        **kwargs: Passed to pkg_tar. See: https://github.com/bazelbuild/rules_pkg/blob/main/docs/0.7.0/reference.md#pkg_tar
+    """
+    if root != None and not root.startswith("/"):
+        fail("root path must start with '/' but got '{root}', expected '/{root}'".format(root = root))
+
+    if kwargs.pop("package_dir", None):
+        fail("use 'root' attribute instead of 'package_dir'.")
+
+    entrypoint_name = "%s_build_tar_entrypoint.py" % name
+
+    write_file(
+        name = "%s.build_tar_entrypoint" % name,
+        out = entrypoint_name,
+        content = [BUILD_TAR],
+        tags = ["manual"],
+    )
+
+    py_binary(
+        name = "%s.build_tar" % name,
+        srcs = [entrypoint_name],
+        main = entrypoint_name,
+        deps = ["@rules_python//python/runfiles"],
+        data = [
+            "@rules_pkg//pkg/private/tar:build_tar",
+        ],
+        tags = ["manual"],
+    )
+
+    common_kwargs = {
+        "tags": kwargs.pop("tags", None),
+        "visibility": kwargs.pop("visibility", None),
+    }
+
+    runfiles_kwargs = dict(
+        common_kwargs,
+        binary = binary,
+        root = root,
+    )
+
+    pkg_tar_kwargs = dict(
+        kwargs,
+        # Be careful with this option. Leave it as is if you don't know what you are doing
```

Comment:
I know what you mean, but this comment is irrelevant. Instead of phrasing it this way, point people to the issue that discusses this.

Reply:
The way it's phrased is not helpful. Please, remove the `Leave it as is ...` sentence to one that points people to the documentation and explains why.

Reply:
Sure.